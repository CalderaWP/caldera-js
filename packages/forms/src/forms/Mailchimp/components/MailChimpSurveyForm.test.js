import React from 'react';

import { render, fireEvent, getByTestId } from 'react-testing-library';

import { MailChimpSurveyForm } from './index';
import {
	surveyQuestion1,
	surveryQuestion2,
	surveyEmailField,
	surveyQuestion1Id,
	surveyQuestionFields,
	surveyQuestionId2,
} from './surveyFields.fixture';

it('Forms', () => {
	expect(
		render(
			<MailChimpSurveyForm
				listId={'a111'}
				onChange={() => {}}
				onBlur={() => {}}
				questions={surveyQuestionFields}
				emailField={surveyEmailField}
				initialQuestionId={surveyQuestion1Id}
			/>
		)
	).toMatchSnapshot();
});
