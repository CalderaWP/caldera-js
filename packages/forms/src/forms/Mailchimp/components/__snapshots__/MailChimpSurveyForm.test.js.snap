// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      
      <div>
        
        <div
          class="caldera"
        >
          <form
            class="mc-subscribe-a111 caldera-form"
          >
            <div
              class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
            >
              <div
                class="caldera-column width-1 sc-bdVaJa iHZvIS"
                id="r1-c1"
              >
                <div
                  class="caldera-field-group caldera-field-area-checkbox caldera-field-wrapper caldera-field-wrapper-checkbox"
                >
                  <input
                    class="caldera-field caldera-field-checkbox"
                    id="fakeGroup2"
                    type="checkbox"
                    value="1"
                  />
                  <label
                    class="caldera-field-label caldera-field-label-checkbox"
                    for="fakeGroup2"
                  >
                    Are you new to JavaScript
                     
                  </label>
                </div>
              </div>
            </div>
            <div
              class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
            >
              <div
                class="caldera-column width-1 sc-bdVaJa iHZvIS"
                id="r2-c1"
              />
              <div
                class="caldera-column width-1 sc-bdVaJa iHZvIS"
                id="r2-c2"
              >
                <div
                  class="caldera-field-group caldera-field-area-submit caldera-field-wrapper caldera-field-wrapper-submit"
                >
                  <label
                    class="caldera-field-label caldera-field-label-submit"
                    for="submitQuestion"
                  >
                    
                     
                  </label>
                  <input
                    class="caldera-field caldera-field-text"
                    id="submitQuestion"
                    type="submit"
                  />
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    
    <div>
      
      <div
        class="caldera"
      >
        <form
          class="mc-subscribe-a111 caldera-form"
        >
          <div
            class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
          >
            <div
              class="caldera-column width-1 sc-bdVaJa iHZvIS"
              id="r1-c1"
            >
              <div
                class="caldera-field-group caldera-field-area-checkbox caldera-field-wrapper caldera-field-wrapper-checkbox"
              >
                <input
                  class="caldera-field caldera-field-checkbox"
                  id="fakeGroup2"
                  type="checkbox"
                  value="1"
                />
                <label
                  class="caldera-field-label caldera-field-label-checkbox"
                  for="fakeGroup2"
                >
                  Are you new to JavaScript
                   
                </label>
              </div>
            </div>
          </div>
          <div
            class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
          >
            <div
              class="caldera-column width-1 sc-bdVaJa iHZvIS"
              id="r2-c1"
            />
            <div
              class="caldera-column width-1 sc-bdVaJa iHZvIS"
              id="r2-c2"
            >
              <div
                class="caldera-field-group caldera-field-area-submit caldera-field-wrapper caldera-field-wrapper-submit"
              >
                <label
                  class="caldera-field-label caldera-field-label-submit"
                  for="submitQuestion"
                >
                  
                   
                </label>
                <input
                  class="caldera-field caldera-field-text"
                  id="submitQuestion"
                  type="submit"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
